betterproto-extras/src/py_any_extras.rs

```diff
@@ -0,0 +1,68 @@
+use crate::error::Result;
+use pyo3::{PyAny, Py, sync::GILOnceCell};
+
+pub trait PyAnyExtras {
+    fn qualified_name(&self) -> Result<String>;
+    fn qualified_class_name(&self) -> Result<String>;
+    fn get_proto_meta(&self) -> Result<&PyAny>;
+    fn get_class(&self, field_name: &str) -> Result<&PyAny>;
+    fn create_instance(&self, field_name: &str) -> Result<&PyAny>;
+    fn is_list_field(&self, field_name: &str) -> Result<bool>;
+    fn oneof_group(&self, field_name: &str) -> Result<Option<String>>;
+    fn py_identifier(&self) -> u64;
+}
+
+impl PyAnyExtras for PyAny {
+    fn qualified_name(&self) -> Result<String> {
+        let module = self.getattr("__module__")?;
+        let name = self.getattr("__name__")?;
+        Ok(format!("{module}.{name}"))
+    }
+
+    fn qualified_class_name(&self) -> Result<String> {
+        self.getattr("__class__")?.qualified_name()
+    }
+
+    fn get_proto_meta(&self) -> Result<&PyAny> {
+        Ok(self.getattr("_betterproto")?)
+    }
+
+    fn get_class(&self, field_name: &str) -> Result<&PyAny> {
+        let cls = self.getattr("cls_by_field")?.get_item(field_name)?;
+        Ok(cls)
+    }
+
+    fn create_instance(&self, field_name: &str) -> Result<&PyAny> {
+        Ok(self.get_class(field_name)?.call0()?)
+    }
+
+    fn is_list_field(&self, field_name: &str) -> Result<bool> {
+        let cls = self.getattr("default_gen")?.get_item(field_name)?;
+        let module = cls.getattr("__module__")?;
+        let name = cls.getattr("__name__")?;
+        Ok(module.to_string() == "builtins" && name.to_string() == "list")
+    }
+
+    fn oneof_group(&self, field_name: &str) -> Result<Option<String>> {
+        let opt = self
+            .getattr("oneof_group_by_field")?
+            .call_method1("get", (field_name,))?
+            .extract()?;
+        Ok(opt)
+    }
+
+    fn py_identifier(&self) -> u64 {
+        static FUN_CACHE: GILOnceCell<Py<PyAny>> = GILOnceCell::new();
+        let py = self.py();
+        let fun = FUN_CACHE.get_or_init(py, || {
+            let fun = py
+                .eval("id", None, None)
+                .expect("This is a valid Python expression");
+            Py::from(fun)
+        });
+        fun.call1(py, (self,))
+            .expect("Identity function is callable")
+            .extract::<u64>(py)
+            .expect("Identity function always returns an integer")
+    }
```

Comment:
```suggestion
    fn py_identifier(&self) -> u64 {
        return self.as_ptr() as usize;
    }
```
Does this not work here?

Reply:
Indeed, this seems to be identically to Python`s buildin `id` function.  Thanks for pointing it out!